src/stan/model/indexing/lvalue.hpp

```diff
@@ -45,7 +45,8 @@ inline void assign(T& x, const nil_index_list& /* idxs */, U y,
  * @param[in] depth Indexing depth (default 0; ignored
  */
 template <typename T, typename U,
-          typename = require_all_not_stan_scalar_t<U, T>>
+          typename = require_all_not_stan_scalar_t<U, T>,
+          typename = require_all_not_std_vector_t<U, T>>
```

Comment:
This overload can still be used for std vectors with same scalar type.
```suggestion
          typename = require_any_t<is_not_std_vector_t<U, T>, is_same_vt<U, T>>>
```